app/Providers/ErrorReportingProvider.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * ErrorReportingProvider.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @link       https://www.librenms.org
+ *
+ * @copyright  2022 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+namespace App\Providers;
+
+use ErrorException;
+use Facade\FlareClient\Report;
+use Facade\Ignition\Facades\Flare;
+use Illuminate\Support\Str;
+use LibreNMS\Config;
+use LibreNMS\Util\Git;
+use LibreNMS\Util\Version;
+
+class ErrorReportingProvider extends \Facade\Ignition\IgnitionServiceProvider
+{
+    protected $errorReportingLevel = E_ALL & ~E_NOTICE;
+    private $laravelErrorHandler;
+
+    public function boot()
+    {
+        // don't report when:
+        Flare::filterExceptionsUsing(function (\Exception $e) {
```

Comment:
You switched from `filterReportsUsing` to `filterExceptionsUsing`, which indeed happens earlier, but it won't catch log-message logging (if we would enable that).

Maybe we should convert these checks to a static function, and set both  `filterReportsUsing` and `filterExceptionsUsing`, for good measure :)

Reply:
There is a bug with filterReportsUsing , the flare client will think it sent a report and generate an UUID, will fix that upstream some day.. so using filterExceptionsUsing is good also :P

Reply:
Setting both seems fine to me.  From what I could sus of the flow filterExceptionsUsing is always called currently.

Reply:
It's not called if it's a log message (that feature is disabled tho)